src/main/java/net/minecraftforge/event/anvil/AnvilDamageEvent.java

```diff
@@ -0,0 +1,87 @@
+package net.minecraftforge.event.anvil;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.world.World;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+@Cancelable
+public class AnvilDamageEvent extends Event
```

Comment:
Why does this event not include the world and position of the anvil?

Reply:
The general purpose catch-all event is called from "AnvilBlock#damage" which has and provides no contexts outside of the previous Blockstate. This event is just as stated a "general catch all" event and provides an entry point for anyone who wants to just stop anvil damage without context. Falling and Container sub-events are the only ones that can and does provide additional context.

I've adressed your points in regard to missing contexts such as:
Falling now has the FallingBlockEntity Context and World Context.
Container now have the RepairContainer Context.

Reply:
Is that event even useful at all then? There is no context at all...

Reply:
I mean that's fair enough.
I'll remove it then and just keep falling and container.